app/src/test/groovy/pl/futurecollars/invoicing/controller/InvoiceControllerIntegrationTest.groovy

```diff
@@ -18,7 +15,7 @@ import static pl.futurecollars.invoicing.TestHelpers.invoice
 
 @SpringBootTest
 @AutoConfigureMockMvc
-class InvoiceControllerIntegrationTest extends Specification {
+class InvoiceControllerIntegrationTest extends HelperController {
 
     @Autowired
     private MockMvc mockMvc
```

Comment:
dodałeś w tej klasie dziedziczenie po klasie pomocniczej, ale nie wykorzystujesz jej w pełni, np masz teraz w 2 miejscach zmienną ENDPOINT, jeśli jest ona w klasie nadrzędnej to możesz z niej skorzystać w klasie dziedziczącej. Przejrzyj proszę te testy i powyciągaj części wspólne, jeśli już podjąłeś decyzje o klasie pomocniczej